cmd/query.go

```diff
@@ -85,6 +87,31 @@ $ %s q ibc-denoms ibc-0`,
 	return cmd
 }
 
+func queryBasefromIBC(a *appState) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "denom-trace [part1] [denom-hash]",
+		Short: "query that take the base denominaton from the IBC denomination trace",
+		Args:  cobra.ExactArgs(2),
+		Example: strings.TrimSpace(fmt.Sprintf(`
+$ %s query denom-trace osmosis-0 9BBA9A1C257E971E38C1422780CE6F0B0686F0A3085E2D61118D904BFE0F5F5E
+$ %s q denom-trace osmosis-0 9BBA9A1C257E971E38C1422780CE6F0B0686F0A3085E2D61118D904BFE0F5F5E`,
+			appName, appName,
+		)),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			res, err := chain.ChainProvider.QueryDenomTrace(cmd.Context(), args[1])
```

Comment:
This doesn't compile because `ChainProvider` is an abstraction on a `Chain` struct not in a package called chain. 
You have to use the `chain-id` argument to the command to get the chain from the relayer's config.